packages/satcheljs-react/test/reactiveTests.tsx

```diff
@@ -0,0 +1,91 @@
+import 'jasmine';
+import {createStore, action} from 'satcheljs';
+import * as React from 'react';
+import reactive from '../lib/reactive';
+
+let sequenceOfEvents: any[];
+
+describe("reactive decorator", () => {
+    it("injects subtree as props for classical components", () => {
+        let store = createStore("testStore", {
+            foo: "value"
+        });
+
+        @reactive({
+            foo: () => store.foo
+        })
+        class TestComponent extends React.Component<any, any> {
+            render() {
+                let {foo, hello} = this.props;
+                expect(foo).toBe(store.foo);
+                expect(hello).toBe('world');
+                return <div>{foo}</div>;
+            }
+        }
+
+        let comp = new TestComponent({hello: 'world'});
+        comp.render();
+    });
+
+    it("allows classical components to be tested in a pure manner", () => {
```

Comment:
This behavior is by design - the passed in props takes precedence over the reactive decorator.